libbeat/docs/processors-using.asciidoc

```diff
@@ -513,6 +514,45 @@ section.
 NOTE: If you define an empty list of fields under `include_fields`, then only
 the required fields, `@timestamp` and `type`, are exported.
 
+[[rename-fields]]
+=== Rename fields from events
+
+The `rename_fields` processor specifies a list of fields that should be renamed.
+The list of fields to be renamed are listed under `fields`. Each entry contains a `from: old-key` and a `to: new-key` pair. `from` is the origin and `to` the target name of the field.
+
+The renaming of fields can be useful in case two fields potentially conflict.
+Having a field with the name `a:2` and `a.b:4` will return an error from
+Elasticsearch during ingest time as `a` would be an integer and object at the
+same time. To prevent this the rename processor can be used to rename `a` to
+`a.value` for example.
```

Comment:
I was struggling on "Having a field with the name `a:2` and `a.b:4`" so I have a suggestion for the paragraph.

Renaming fields can be useful in cases where field names cause conflicts. For example if an event has two fields `a` and `a.b` that are both assigned a scalar values (e.g. `{"a": 1, "a.b": 2}`) this will result in an error at ingest time from Elasticsearch. This is because the value of `a` cannot simultaneously be a scalar and an object. To prevent this `rename_fields` can be used to rename `a` to `a.value`.

Reply:
Done, thanks for the proposal.